fe/fe-core/src/main/java/com/starrocks/transaction/ExplicitTxnState.java

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.transaction;
+
+import com.starrocks.sql.ast.DmlStmt;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Explicit transaction in a session. The temporary state generated by multiple statements in a transaction is recorded in
+ * ExplicitTxnStateItem, and the transaction state is recorded in TransactionState.
+ */
+public class ExplicitTxnState {
+    private TransactionState transactionState;
+    private final List<ExplicitTxnStateItem> explicitTxnStateItems = new ArrayList<>();
+
+    public ExplicitTxnState() {
+    }
+
+    public TransactionState getTransactionState() {
+        return transactionState;
+    }
+
+    public void setTransactionState(TransactionState transactionState) {
+        this.transactionState = transactionState;
+    }
+
+    public List<ExplicitTxnStateItem> getTransactionStateItems() {
+        return explicitTxnStateItems;
+    }
+
+    public void addTransactionItem(ExplicitTxnStateItem explicitTxnStateItem) {
+        explicitTxnStateItems.add(explicitTxnStateItem);
+    }
+
+    public static class ExplicitTxnStateItem {
+        private long loadedRows;
+        private long loadedBytes;
+        private long filteredRows;
+
+        private List<TabletCommitInfo> tabletCommitInfos = null;
+        private List<TabletFailInfo> tabletFailInfos = null;
+
+        private DmlStmt dmlStmt;
```

Comment:
what this member for?

Reply:
Because the subsequent process needs to determine whether a specific function needs to be called based on the type of DML